src/base/color-mixin.ts

```diff
@@ -14,174 +16,176 @@ import {ColorCommonInstance} from 'd3-color';
  * @returns {ColorMixin}
  */
 // tslint:disable-next-line:variable-name
-export const ColorMixin = Base => class extends Base {
-    public _colors;
-    public _colorAccessor: (d, i) => any;
-    public _colorCalculator;
+export function ColorMixin<TBase extends Constructor<BaseMixin>>(Base: TBase) {
+    return class extends Base {
+        public _colors;
+        public _colorAccessor: (d, i) => any;
+        public _colorCalculator;
 
-    constructor () {
-        super();
+        constructor(...args: any[]) {
```

Comment:
does it take args?

Reply:
Typescript was giving error. I am planning to relook.

Reply:
It seems it is needed otherwise it gives following error:

```
error TS2545: A mixin class must have a constructor with a single rest parameter of type 'any[]'.
```